acme_diags/driver/tc_analysis_driver.py

```diff
@@ -0,0 +1,541 @@
+            mod_mon.append(mon[0])
+            mod_wnd.append(np.max(wind))
+
+            wind_new = wind[wind > 35]
+            mod_ace[np.where(years - yer[0] == 0)] = (
+                mod_ace[np.where(years - yer[0] == 0)] + np.sum(wind_new ** 2) / 1e4
+            )
+
+    mod_vars = {
+        "mod_mon": mod_mon,
+        "mod_wnd": mod_wnd,
+        "mod_num": mod_num,
+        "mod_num_ocn": mod_num_ocn,
+        "mod_ace_mean": np.mean(mod_ace),
+    }
+    return mod_vars
+
+
+def generate_tc_metrics_from_obs_files(reference_data_path: str) -> Dict[str, Any]:
+    """Generates tropical cyclone metrics from observation files.
+
+    :param reference_data_path: Reference data path
+    :type reference_data_path: str
+    :return: TC Metrics
+    :rtype: Dict[str, Any]
+
+    # TODO: Add tests to cover this function
+    """
+    print("\nGenerating TC Metrics from Obs Files")
+    print("======================================")
+    # Using IBTrACS data, store a tuple for each basin
+    # (mean ace, tc_intensity_dist, seasonal_cycle, # observed hurricane per year,
+    # # of storms over the ocean)
+    result_obs: Dict[str, Any] = {}
+    result_obs["num_years"] = OBS_YEARS.size
+
+    for basin, basin_info in BASIN_DICT.items():
+        nc = netcdffile(
+            os.path.join(reference_data_path, "IBTrACS.{}.v04r00.nc".format(basin))
+        )
+
+        # Extract and parse variables from .nc file
+        vsmc = np.squeeze(nc["wmo_wind"][:, :])
+        time = np.squeeze(nc["time"][:, :])
+        monthmc, yearic = _get_monthmc_yearic(time)
+        num_rows = time.shape[0]
+
+        # Calculate mean ace
+        mean_ace = _calc_mean_ace(vsmc, yearic, num_rows)
+
+        # Calculate TS intensity distribution and seasonal cycle
+        mon, wnd = _get_mon_wind(vsmc, monthmc, yearic, num_rows)
+        tc_intensity_dist = _calc_ts_intensity_dist(wnd)
+        pdf_obs_seasonal_cycle = _calc_seasonal_cycle(mon)
+
+        # Number of hurricanes and storms over the ocean
+        num_hurricanes_per_yr = basin_info[5]
+        num_storms_over_ocean = 1
+
+        # Final result
+        result_obs[basin] = (
+            mean_ace,
+            tc_intensity_dist,
+            pdf_obs_seasonal_cycle,
+            num_hurricanes_per_yr,
+            num_storms_over_ocean,
+        )
+
+    return result_obs
+
+
+def _get_monthmc_yearic(time: "MaskedArray") -> Tuple[np.ndarray, np.ndarray]:
+    """Extracts the monthmc and yearic by parsing the time variable.
+
+    All masked (missing) data points are ignored.
+
+    :param time: Time variable, units are days since 1858-11-17 00:00:00
+    :type time: MaskedArray
+    :return: Arrays for the months and years based on the day of a hurricane
+    :rtype: Tuple[np.ndarray, np.ndarray]
+    """
+    monthmc = np.zeros(time.shape)
+    yearic = np.zeros(time.shape[0])
+
+    # The units attribute of the time variable
+    days_since = datetime(1858, 11, 17, 0, 0, 0)
```

Comment:
Added a variable to denote that this is the `unit` attribute from the `time` variable of the netcdf file, otherwise it is hard to tell where this came from